pandas/core/frame.py

```diff
@@ -3711,13 +3713,7 @@ def extract_unique_dtypes_from_dtypes_set(
             extracted_dtypes = [
                 unique_dtype
                 for unique_dtype in unique_dtypes
-                # error: Argument 1 to "tuple" has incompatible type
-                # "FrozenSet[Union[ExtensionDtype, str, Any, Type[str],
-                # Type[float], Type[int], Type[complex], Type[bool]]]";
-                # expected "Iterable[Union[type, Tuple[Any, ...]]]"
-                if issubclass(
-                    unique_dtype.type, tuple(dtypes_set)  # type: ignore[arg-type]
-                )
+                if np_issubclass_compat(unique_dtype, dtypes_set)
```

Comment:
i don't think we need to create yet another method here. is there a reason you cannot just use something like

```
return issubclass(unique_dtype.type, tuple(dtypes_set)) or (
        np.number in dtypes_set and is_numeric_dtype(unique_dtype))
    )
```

Reply:
no reason. Done

Reply:
Having now tried this I think the reason might be mypy
```
pandas/core/frame.py:3715: error: Argument 1 to "tuple" has incompatible type "FrozenSet[Union[ExtensionDtype, str, Any, Type[str], Type[float], Type[int], Type[complex], Type[bool], Type[object]]]"; expected "Iterable[Union[type, Tuple[Any, ...]]]"  [arg-type]

```

Reply:
Reverted to having a separate method

Reply:
just ignore mypy here, this makes groking the code way more complicated.

Reply:
ok!